internal/logger/logger_test.go

```diff
@@ -39,6 +45,114 @@ func BenchmarkLogger(b *testing.B) {
 	})
 }
 
+// mockKeyValues will return a slice of alternating keys and values of the
+// given length and the resulting JSON string.
+func mockKeyValuesB(b *testing.B, length int) KeyValues {
+	b.Helper()
+
+	keysAndValues := KeyValues{}
+	for i := 0; i < length; i++ {
+		keyName := fmt.Sprintf("key%d", i)
+		valueName := fmt.Sprintf("value%d", i)
+
+		keysAndValues.Add(keyName, valueName)
+	}
+
+	return keysAndValues
+}
+
+func BenchmarkIOSinkInfo(b *testing.B) {
+	keysAndValues := mockKeyValuesB(b, 100)
+
+	b.ReportAllocs()
+	b.ResetTimer()
+
+	sink := NewIOSink(bytes.NewBuffer(nil))
+
+	for i := 0; i < b.N; i++ {
+		sink.Info(0, "foo", keysAndValues...)
+	}
+}
+
+func mockKeyValuesT(t *testing.T, length int) (KeyValues, map[string]interface{}) {
+	t.Helper()
```

Comment:
The call to `t.Helper()` here is unnecessary because there are no calls to `t` methods that need to know where in the call stack to find the code line number (e.g. `t.Error`). Remove `t *testing.T` from the function parameters and merge this function with `mockKeyValuesB`.